ext/ffi_c/Function.c

```diff
@@ -215,6 +215,52 @@ async_cb_dispatcher_set(struct async_cb_dispatcher *ctx)
     async_cb_dispatcher = ctx;
 }
 #endif
+
+static void
+async_cb_dispatcher_initialize(struct async_cb_dispatcher *ctx)
+{
+        ctx->async_cb_list = NULL;
+
+#if !defined(_WIN32)
+        /* n.b. we _used_ to try and destroy the mutex/cond before initializing here,
+         * but it's undefined what happens if you try and destory an unitialized cond.
+         * glibc in particular seems to wait for any concurrent waiters to finish before
+         * destroying a condvar, trying to destroy a condvar after fork that someone was
+         * waiting on pre-fork won't work. Just re-init he memory directly. */
+        pthread_mutex_init(&ctx->async_cb_mutex, NULL);
+        pthread_cond_init(&ctx->async_cb_cond, NULL);
```

Comment:
Minor: I wonder if, given the comment, it would be cleaner to use

```c
ctx->async_cb_mutex = PTHREAD_MUTEX_INITIALIZER;
ctx->async_cb_cond = PTHREAD_COND_INITIALIZER;
```

Reply:
I guess they're equivalent, yeah. I can change this if we prefer.